spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/TaskController.java

```diff
@@ -125,6 +125,10 @@ public TaskController(TaskDefinitionRepository repository, ArtifactRegistry regi
 	@RequestMapping(value = "/definitions", method = RequestMethod.POST)
 	public void save(@RequestParam("name") String name,
 			@RequestParam("definition") String dsl) {
+		TaskDefinition previous = repository.findOne(name);
+		if (previous != null) {
+			throw new ResourceAlreadyExistsException(name, "task definition");
+		}
```

Comment:
Why are we making the repositories dumb to support this change?  Why can't we keep the repository defensive (have them responsible for maintaining the integrity of their data) and have the Controller catch exceptions thrown and translate them appropriately for the HTTP context?